src/bsoncxx/config/compiler.hpp

```diff
@@ -12,4 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#if defined(_MSC_VER)
+
+#pragma warning(push)
+#pragma warning(disable: 4251 4273 4275)
```

Comment:
Thanks for making me take another look at this. I think the only one we really need is 4251, and I've added a comment about that here and for mongocxx. I think the other ones were spurious due to misconfigurations addressed elsewhere in this PR.